### PR TITLE
feat(playback): merge anni-player into anni-playback

### PR DESCRIPTION
**Pull request by snylonue**

### CODE REVIEW

File: anni-common/src/models.rs [hunk 1]
```diff
@@ -38,13 +44,59 @@ impl<'a> Clone for RawTrackIdentifier<'a> {
...
+}
+
+impl Clone for TrackIdentifier {
+    fn clone(&self) -> Self {
+        self.inner.to_owned()
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct ParseError;
```
Comment (Yesterday17) [inline anni-common/src/models.rs:94]: Use `thiserror` instead?

File: anni-common/src/models.rs [hunk 2]
```diff
@@ -26,6 +28,10 @@ impl<'a> RawTrackIdentifier<'a> {
             },
         }
     }
+
+    pub fn copied(&'a self) -> Self {
```
Comment (Yesterday17) [inline anni-common/src/models.rs:32]: If it compiles after removing the `Clone` implementation, just replace it :)

File: anni-common/src/models.rs (hunk 1 above)
Comment (snylonue) [reply anni-common/src/models.rs:94]: Is it too heavy since it is just a trivial error type?

Comment (Yesterday17) [reply anni-common/src/models.rs:94]: We may separate the error to somewhat like `invalid album id`, `invalid disc id` and `invalid track id`

Comment (snylonue) [reply anni-common/src/models.rs]: updated

File: anni-common/src/models.rs (hunk 2 above)
Comment (snylonue) [reply anni-common/src/models.rs:34]: Actually it is hard because clone accepts arbitrary lifetime of self but we want to return a `'a`

**Review by Yesterday17 — APPROVED** [review]

**Pull request merged**